File: Docs/Stripper/cs/StripperContext.cs
using System;

namespace smc
{
	public class StripperContext
	{
		public void FSMError(String s , object o){}

		public void OutsideIn() 
		{
			Console.WriteLine("OutsideIn");
		}
		public void StartingSlashOut()
		{
			Console.WriteLine("StartingSlashOut");
		}
		public void StartingSlashIn() 
		{
			Console.WriteLine("StartingSlashIn");
		}
		public void StartingStarIn() 
		{
			Console.WriteLine("StartingStarIn");
		}
		public void StartingStarOut()
		{
			Console.WriteLine("StartingStarOut");
		}
		public void InCommentIn() 
		{
			Console.WriteLine("InCommentIn");
		}
		public void InCommentOut() 
		{
			Console.WriteLine("InCommentOut");
		}
		public void PutSlash()
		{
			Console.WriteLine("PutSlash");
		}
		public void PutChar()
		{
			Console.WriteLine("PutChar");
		}
		public void StarAfterSlashIn()
		{
			Console.WriteLine("StarAfterSlashIn");
		}
		public void StarAfterSlashOut() 
		{
			Console.WriteLine("StarAfterSlashOut");
		}
		public void SecondSlashIn()
		{
			Console.WriteLine("SecondSlashIn");
		}
		public void SecondSlashOut() 
		{
			Console.WriteLine("SecondSlashOut");
		}
		public void OutsideOut() 
		{
			Console.WriteLine("OutsideOut");
		}
		
		public static void Main()
		{
			StripFSM fsm = new StripFSM();
			fsm.Star();
		}
		
	}
}

File: Docs/Stripper/java/StripperContext.java
package smc.stripper;

import java.io.InputStream;
import java.io.OutputStream;
import java.io.IOException;


public class StripperContext
{
    private int itsChar;
    private InputStream itsInStream;
    private OutputStream itsOutStream;

    public StripperContext()
    {}

    public void SetStreams(InputStream iStream, OutputStream oStream)
    {
      itsInStream = iStream;
      itsOutStream = oStream;
    }

    public int ReadChar()
    {
      int c = -1;
      try
      {
        c = itsInStream.read();
        itsChar = c;
      }
      catch (IOException ioException)
      {
        System.err.println("Read - IOExeception");
      }
      return c;
    }

    public void PutChar()
    {
      try
      {
        itsOutStream.write(itsChar);
      }
      catch (IOException exception)
      {
        System.err.println("PutChar - IOExeception");
      }
    }

    public void PutSlash()
    {
      try
      {
        itsOutStream.write('/');
      }
      catch (IOException exception)
      {
        System.err.println("PutSlash - IOExeception");
      }
    }

    public void OutsideIn()
    {
      System.err.println("OutsideIn");
    }

    public void OutsideOut()
    {
      System.err.println("OutsideOut");
    }

    public void StartingSlashIn()
    {
      System.err.println("StartingSlashIn");
    }

    public void StartingSlashOut()
    {
      System.err.println("StartingSlashOut");
    }

    public void SecondSlashIn()
    {
      System.err.println("SecondSlashIn");
    }

    public void SecondSlashOut()
    {
      System.err.println("SecondSlashOut");
    }

    public void InCommentIn()
    {
      System.err.println("InCommentIn");
    }

    public void InCommentOut()
    {
      System.err.println("InCommentOut");
    }

    public void StarAfterSlashIn()
    {
      System.err.println("StarAfterSlashIn");
    }

    public void StarAfterSlashOut()
    {
      System.err.println("StarAfterSlashOut");
    }

    public void StartingStarIn()
    {
      System.err.println("StartingStarIn");
    }

    public void StartingStarOut()
    {
      System.err.println("StartingStarOut");
    }

    public void FSMError(String t, String s)
    {
      System.err.println("Transition error: " + t + " in state " + s );
    }
   
}
